app/plugins/furaffinity.ts

```diff
@@ -18,22 +19,53 @@ export class FuraffinityPlugin extends BaseSitePlugin {
             return Promise.resolve(null);
         }
 
-        // FA download URLs look like so:
-        // http://d.facdn.net/art/[username]/[id]/[id].[username]_[origfilename].[ext]	
-        // Preivew URLs look like so:
+        // FA download URLs vary a bit depending on the submission type:
+        //
+        // art   https://d.facdn.net/art/[username]/[id1]/[id2].[username]_[origfilename].[ext]
+        // music https://d.facdn.net/art/[username]/music/[id1]/[id2].[username]_[origfilename].[ext]
+        // story https://d.facdn.net/art/[username]/stories/[id1]/[id2].[username]_[origfilename].[ext]
+        //
+        // id1 and id2 can be slightly different numbers.  In limited
+        // testing, I (Eupeptic) think id1 and id2 are equal if the artist
+        // uploaded the submission and then never updated it.  If the
+        // artist does update the sumbission, id2 stays the same, but id1
+        // changes to a larger number.  I've seen id1 be as much as 475
+        // and as little as 1 larger than id2.  I'm not sure if id1 can
+        // be smaller than id2, but I think this is unlikely.  I have seen
+        // them differ on files uploaded before the new FA UI on
+        // 2019-12-31, so I don't think it's related to the UI change.
+
+        // Preview URLs look like this:
         // //t.facdn.net/22795737@400-[id].[ext]
+        // Yes, the protocol is missing.
+        // They don't seem to vary by submission type.
+        // The 400- can also be 200- or 800- , which probably sets the
+        // thumbnail size.
+
         let urlParts = url.split("/");
         let serviceFilename = urlParts[urlParts.length - 1];
         let { filename, ext } = getFilenameParts(serviceFilename);
+        // Take id2 here.  Since it's part of the filename on FA,
+        // hopefully that's the right choice.
```

Comment:
My thinking on this is that it's the submission that the user is saving, not a specific iteration of that submission. I'm thinking it makes more sense to pick the number that matches the submission ID in the page URL, since that's what people expect and what people will use if they want to take a file and find the submission again on FA.

Reply:
It's OK with me to use either id1 or id2.  After I wrote that comment, I figured that was a policy decision that you were better placed to make.